### PR TITLE
Auto merge patch dependency updates

### LINKED ISSUE #10295: Auto merge green dependabot PRs
**Description**

At the moment, we have to manually merge passing dependabot PRs. While there is always a risk to do so, we could at the very least enable auto merging patch releases (assuming the PR is green). I think in most cases the risk is quite low.

There are cases where dependencies do not use semantic versioning, so we can't rely on the meaning of the patch release. But in general, we rarely do more checks if our PRs are green, so I'm not sure what we gain by keeping them on a manual lifecycle.

[GitHub has some docs on how to automate merging PRs](https://docs.github.com/en/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions#enable-auto-merge-on-a-pull-request):

```yaml
name: Dependabot auto-merge patch updates
on: pull_request

permissions:
  contents: write
  pull-requests: write

jobs:
  dependabot:
    runs-on: ubuntu-latest
    if: ${{ github.actor == 'dependabot[bot]' }}
    steps:
      - name: Dependabot metadata
        id: metadata
        uses: dependabot/fetch-metadata@v1.1.1
        with:
          github-token: "${{ secrets.GITHUB_TOKEN }}"
      - name: Enable auto-merge for Dependabot PRs
        if: ${{ steps.metadata.outputs.update-type == 'version-update:semver-patch' }}
        run: gh pr comment "${PR_ID}" --body "bors merge"
        env:
          PR_ID: ${{github.event.pull_request.number}}
          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}

```

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR adds a workflow which runs on every dependabot PRs, and will automatically merge the PR iff the checks pass, and it is a patch version (according to dependabot). Merging is still  done via bors.

The workflow is made of three steps: checking the dependabot metadata (used to determine the type of updates, e.g. patch, minor, etc.), waiting for specific checks to pass, then approving the PR with a `bors merge` comment (triggering the merge).

If bors fails to merge, or checks fail, then manual action is still required.

## Related issues

closes #10295

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by npepinpe** [comment]
So all good, but it seems the GitHub bot is not an approved review ;)

**Comment by npepinpe** [comment]
For testing, see this run: https://github.com/camunda/zeebe/actions/runs/3045213733/jobs/4906497130

**Comment by npepinpe** [comment]
One last thing, I wasn't sure how to test the dependabot part without a dependabot PR, so we will find out on the first dependabot PR.

File: .github/workflows/auto-merge-deps.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+#
+# It will merge the PR only if: it is authored by dependabot[bot], is a patch semantic update, and
+# all CI checks are successful (ignoring the soon-to-be-removed Jenkins check).
+#
+# The workflow is divided into multiple sequential jobs to allow giving only minimal permissions to
+# the GitHub token passed around.
+#
+# Once we're using the merge queue feature, I think we can simplify this workflow a lot by relying
+# on dependabot merging PRs via its commands, as it will always wait for checks to be green before
+# merging.
```
Comment (korthout) [inline .github/workflows/auto-merge-deps.yml:12]: 👍 Nice

File: .github/workflows/auto-merge-deps.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+jobs:
+  auto-merge:
+    name: Auto-merge dependabot PRs
+    runs-on: ubuntu-latest
+    if: ${{ github.repository == 'camunda/zeebe' && github.actor == 'dependabot[bot]' }}
+    steps:
+      - uses: actions/checkout@v3
+      - id: metadata
+        name: Fetch dependency metadata
+        if: false
```
Comment (korthout) [inline .github/workflows/auto-merge-deps.yml:35]: ❓ `if: false`, so this step is always skipped?

**Review by korthout — APPROVED** [review]
LGTM 👍 

👏 Nice work @npepinpe 🚀 I'm excited to see this work

File: .github/workflows/auto-merge-deps.yml (hunk 2 above)
Comment (npepinpe) [reply .github/workflows/auto-merge-deps.yml:35]: No, that's a left over for testing, my bad :see_no_evil:

**Comment by npepinpe** [comment]
bors merge

**Comment by korthout** [comment]
Nice to see it in action @npepinpe https://github.com/camunda/zeebe/pull/10358

**Pull request merged**